### PR TITLE
Data Terminals are now buildable

### PR DESCRIPTION
**Pull request by francinum**
Allows construction and destruction of data terminals, and fixes phone terminal link messages.

## Changelog

:cl:
add: Data Terminals are buildable, you can get them from the Engineering Fabricator.
fix: phone connection feedback messages no longer error on trying to connect to a terminal it's already connected to.
/:cl:

### CODE REVIEW

File: code/modules/power/data_terminal.dm [hunk 1]
```diff
@@ -64,3 +92,40 @@
...
+	if(HAS_TRAIT(src, TRAIT_NODROP) && loc == user)
+		to_chat(user, span_warning("\The [src] is stuck to your hand!"))
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+	var/turf/T = get_turf(src)
+	if(T.underfloor_accessibility < UNDERFLOOR_INTERACTABLE && isfloorturf(T))
+		to_chat(user, span_warning("You can only install the [src] if the floor plating is removed!"))
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+	user.visible_message(
+		"[user] starts screwing [src] into \the [T].",
+		"you start screwing [src] into \the [T]",
```
Comment (francinum) [inline code/modules/power/data_terminal.dm:121]:
```suggestion
		"You start screwing [src] into \the [T]",
```

File: code/modules/power/data_terminal.dm [hunk 2]
```diff
@@ -64,3 +92,40 @@
...
+		"You hear quiet metal scraping.")
+	tool.play_tool_sound(src, 50)
+	if(!do_after_mob(user, src, 10 SECONDS, DO_PUBLIC))
+		to_chat(user, span_warning("You need to stand still to install [src]!"))
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+	playsound(src.loc, 'sound/machines/click.ogg', 75, TRUE)
+	user.visible_message("[user] installed \the [src] to \the [T]", "You installed \the [src] to \the [T]")
```
Comment (francinum) [inline code/modules/power/data_terminal.dm:128]:
```suggestion
	user.visible_message("[user] installed \the [src] on \the [T]", "You installed \the [src] on \the [T]")
```

File: code/modules/power/data_terminal.dm [hunk 3]
```diff
@@ -12,9 +12,35 @@
...
+/obj/machinery/power/data_terminal/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(.)
+		return
+	/// Oop, we have a connected machine, make sure they're sure.
+	if(connected_machine && !(tgui_alert(user, "The link light is on, deconstructing this will disconnect the device.", "Disconnect?",list("Disconnect","Abort")) == "Disconnect"))
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+	user.visible_message(
+		"[user] starts unscrewing [src] from \the [src.loc].",
+		"you start unscrewing [src] from \the [src.loc]",
```
Comment (francinum) [inline code/modules/power/data_terminal.dm:32]:
```suggestion
		"You start unscrewing [src] from \the [src.loc]",
```

File: code/modules/power/data_terminal.dm [hunk 4]
```diff
@@ -12,9 +12,35 @@
...
 
+/obj/machinery/power/data_terminal/Destroy()
+	. = ..()
+	// Disconnect from the seizing machine.
+	disconnect_machine(connected_machine)
+
 /obj/machinery/power/data_terminal/should_have_node()
 	return TRUE
 
+/obj/machinery/power/data_terminal/screwdriver_act(mob/living/user, obj/item/tool)
```
Comment (Kapu1178) [inline code/modules/power/data_terminal.dm:23]: New lines please

File: code/modules/power/data_terminal.dm [hunk 5]
```diff
@@ -64,3 +92,40 @@
...
+	name = "disconnected data terminal"
+	desc = "A data terminal, used for connecting to powerline networks."
+	icon_state = "dterm_construct"
+	icon = 'icons/obj/power.dmi'
+
+/obj/item/data_terminal_construct/examine(mob/user)
+	. = ..()
+	. += span_notice("You can [span_bold("screw")] it to the floor to install it.")
+
+/obj/item/data_terminal_construct/screwdriver_act(mob/living/user, obj/item/tool)
```
Comment (Kapu1178) [inline code/modules/power/data_terminal.dm:108]: new lines please

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Pull request merged**